src/app/app.component.ts

```diff
@@ -1,276 +1,338 @@
     }
-    this.showRefreshLoader = true;
-    for (const mbx of this.mailboxes) {
-      await this.messageCount(mbx);
-    }
-    this.showRefreshLoader = false;
+
+    await this.activeMailbox.uploadFileChunk(this.relayURL, uploadID, binary, 0, 1, skey);
+    this.showBadge(UIAction.fileSent);
+    await this.refreshCounter();
   }
 
-  checkAll(event: { target: HTMLInputElement }) {
-    this.activeMailbox.messages.map(message => message.isSelected = event.target.checked);
+  /**
+   * Download the file and trigger browser's download capability
+   */
+  async downloadFile(message: ZaxFileMessage): Promise<void> {
+    await this.activeMailbox.connectToRelay(this.relayURL);
+    const chunk = await this.activeMailbox.downloadFileChunk(this.relayURL,
+      message.uploadID, 0, message.data.skey);
+
+    // DOM trick to let browser start downloading the binary file
+    const url = window.URL.createObjectURL(new Blob([chunk]));
+    const link = document.createElement('a');
+    link.href = url;
+    // Use original file name
+    link.setAttribute('download', message.data.name);
+    document.body.appendChild(link);
+    link.click();
   }
 
-  async messageCount(mailbox) {
-    await mailbox.connectToRelay(this.relay);
-    const count = await mailbox.relayCount(this.relay);
-    mailbox.messageCount = count;
+  // -------------------------
+  // UI helpers
+  // -------------------------
+
+  selectAllMessages(target: HTMLInputElement): void {
+    this.activeMailbox.messages.map(message => message.isSelected = target.checked);
   }
 
-  async getMessages(mailbox) {
-    this.showMessagesLoader = true;
-    const messages = await mailbox.getRelayMessages(this.relay);
-    mailbox.messages = [];
-    for (const msg of messages) {
-      if (msg.kind === 'file') {
-        msg.data = '📎 File [uploadID: ' + JSON.parse(msg.data).uploadID + ']';
-      }
-      mailbox.messages.push(msg);
-    }
-    this.showMessagesLoader = false;
+  hasSelectedMessages(): boolean {
+    return !!this.activeMailbox.messages?.find(message => message.isSelected);
   }
 
-  async sendMessage(mailbox, form: NgForm) {
-    const { recipient, message } = form.controls;
-    await mailbox.sendToVia(recipient.value, this.relay, message.value);
-    this.messageSent = true;
+  private showBadge(action: UIAction) {
+    this.UIFlags[action] = true;
     setTimeout(() => {
-      this.messageSent = false;
+      this.UIFlags[action] = false;
     }, 3000);
-    form.reset();
-    this.refreshCounter();
-  }
-
-  async deleteMessages(mailbox) {
-    const noncesToDelete = mailbox.messages.filter(message => message.isSelected)
-      .map(message => message.nonce);
-    await mailbox.connectToRelay(this.relay);
-    await mailbox.relayDelete(noncesToDelete, this.relay);
-
-    mailbox.messages = mailbox.messages.filter(message => !message.isSelected);
-
-    // Update counter without polling server
-    mailbox.messageCount = Object.keys(mailbox.messages).length;
   }
 }
```

Comment:
please move glow specific methods to separate service

Reply:
Ok, let me move it in a separate PR

Reply:
> maybe remove compiled scripts, css and icons from repository?

Compiled app needs to be in the repo to be hosted as Github Pages on https://vault12.github.io/zax-dashboard/